File: trafficcounter-backend/src/main/java/ch/bfh/trafficcounter/service/MeasurementPointService.java
/*
 * Copyright 2023 Manuel Riesen, Sandro Rüfenacht, Sven Trachsel
 *
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */

package ch.bfh.trafficcounter.service;

import ch.bfh.trafficcounter.model.dto.geojson.GeoJsonFeatureCollectionDto;
import ch.opentdata.wsdl.MeasurementSiteRecord;

import java.util.List;


/**
 * Service for measurement points.
 *
 * @author Sven Trachsel
 */
public interface MeasurementPointService {

    /**
     * Processes and persists all measurement points.
     *
     * @param measurementSiteRecords measurement site records
     */
    void processAndPersistMeasurementPoints(List<MeasurementSiteRecord> measurementSiteRecords);

    /**
     * Gets a GeoJSON-List of all registered Measurement Points
     * Only finds active measurement points
     *
     * @return a DTO in GeoJSON format
     */
    GeoJsonFeatureCollectionDto getAllMeasurementPointsGeoJson();


    /**
     * Counts number of active measurement points, zero indicates that no data has been loaded yet
     *
     * @return number of active measurement points
     */
    int getNumberOfMeasurementPoints();
}

File: trafficcounter-frontend/src/app/services/measurement-point/measurement-point.service.ts
/*
 * Copyright 2023 Manuel Riesen, Sandro Rüfenacht, Sven Trachsel
 *
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */

import {Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {GeoJsonFeatureCollectionDto} from "../../api/models/geo-json-feature-collection-dto";
import {MeasurementPointControllerService} from "../../api/services/measurement-point-controller.service";

/**
 * Service for operations related to measurement points.
 */
@Injectable({
  providedIn: 'root'
})
export class MeasurementPointService {

  constructor(private readonly measurementPointControllerService: MeasurementPointControllerService) {
  }

  /**
   * Gets all measurement points from the API as GeoJSON.
   */
  public getAllMeasurementPoints(): Observable<GeoJsonFeatureCollectionDto> {
    return this.measurementPointControllerService.getMeasurementPointsGeoJson();
  }
}
